### PR TITLE
Moving extensions to essentials & improving image methods

### PR DESCRIPTION
**Pull request by ja0b**
This PR contains improvements for the imageCropper extensions and now we moved all the base extensions to the essentials package

### CODE REVIEW

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 1]
```diff
@@ -8,9 +8,28 @@ namespace Cogworks.Umbraco.Essentials.Extensions
...
+            if (!includeRetina)
+            {
+                return imageUrl;
+            }
+
+            var retinaUrl = GetCropUrlRetina(image, cropAlias, width, height, enableWebP, quality);
+
+            if (retinaUrl.HasValue())
+            {
+                return returnOnlyRetina ? retinaUrl : $"{imageUrl} 1x, {retinaUrl} 2x";
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:24]: Move `?` and `:` to new line.
```
return returnOnlyRetina
          ? retinaUrl
           : $"{imageUrl} 1x, {retinaUrl} 2x";
```

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 2]
```diff
@@ -8,9 +8,28 @@ namespace Cogworks.Umbraco.Essentials.Extensions
...
+            }
+
+            return imageUrl;
+        }
+
+        public static string GetCropUrlStandard(this IPublishedContent image, string cropAlias = null, int? width = null, int? height = null, bool enableWebP = false, int? quality = null)
+        {
+            var imageUrl = cropAlias.HasValue() ? image.GetCropUrl($"{cropAlias}") : image.GetCropUrl();
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:32]: Same here

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 3]
```diff
@@ -22,16 +41,21 @@ public static string GetCropUrls(this IPublishedContent image, string cropAlias,
...
-                return imageUrl;
+                return string.Empty;
             }
 
             var retinaUrl = GetRetinaCropUrl(imageUrl);
 
-            return retinaUrl.HasValue()
-                ? $"{imageUrl} 1x, {retinaUrl} 2x"
-                : imageUrl;
+            return retinaUrl.HasValue() ? retinaUrl : imageUrl;
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:58]: Same here

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 4]
```diff
@@ -8,9 +8,28 @@ namespace Cogworks.Umbraco.Essentials.Extensions
 {
     public static class ImageCropperExtensions
     {
-        public static string GetCropUrls(this IPublishedContent image, string cropAlias, int? width = null, int? height = null, bool includeRetina = true, bool enableWebP = false, int? quality = null)
+        public static string GetCropUrls(this IPublishedContent image, string cropAlias = null, int? width = null, int? height = null, bool includeRetina = true, bool enableWebP = false, int? quality = null, bool returnOnlyRetina = false)
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:11]: IMO we should move/split those parameters to new line, they are to long in single line.

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 5]
```diff
@@ -8,9 +8,28 @@ namespace Cogworks.Umbraco.Essentials.Extensions
...
+
+            if (retinaUrl.HasValue())
+            {
+                return returnOnlyRetina ? retinaUrl : $"{imageUrl} 1x, {retinaUrl} 2x";
+            }
+
+            return imageUrl;
+        }
+
+        public static string GetCropUrlStandard(this IPublishedContent image, string cropAlias = null, int? width = null, int? height = null, bool enableWebP = false, int? quality = null)
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:30]: IMO we should move/split those parameters to new line, they are to long in single line.

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs [hunk 6]
```diff
@@ -22,16 +41,21 @@ public static string GetCropUrls(this IPublishedContent image, string cropAlias,
                 imageUrl = UpdateCropSettings(imageUrl, width, height, enableWebP, quality);
             }
 
-            if (!includeRetina)
+            return imageUrl;
+        }
+
+        public static string GetCropUrlRetina(this IPublishedContent image, string cropAlias = null, int? width = null, int? height = null, bool enableWebP = false, int? quality = null)
```
Comment (aochmann) [inline Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:47]: IMO we should move/split those parameters to new line, they are to long in single line.

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs (hunk 4 above)
Comment (aochmann) [reply Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:11]: Or we could think to create some configuration class as parameter that contains those settings there.
```
public static string GetCropUrls(this IPublishedContent image, CropOptions options)
```

File: Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs (hunk 1 above)
Comment (ja0b) [reply Source/Cogworks.Umbraco.Essentials/Extensions/ImageCropperExtensions.cs:24]: @aochmann this is down to preferences, IMO the logic is the same

**Review by aochmann — APPROVED** [review]

**Pull request merged**